modules/katello_activation_key.py

```diff
@@ -0,0 +1,201 @@
+        description:
+            - Organization label that the activation key is in
+        required: false
+    lifecycle_environment:
+        description:
+            - Name of the lifecycle environment
+        required: false
+    content_view:
+        description:
+            - Name of the content view
+        required: false
+'''
+
+EXAMPLES = '''
+- name: "Create katello client activation key"
+  katello_activation_key:
+    username: "admin"
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    name: "Katello Clients"
+    organization: Default Organization
+    lifecycle_environment: "Library"
+    content_view:
+      - name: 'client content view'
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun import entities
+    from nailgun.config import ServerConfig
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+
+class NailGun(object):
+    def __init__(self, server, entities, module):
+        self._server = server
+        self._entities = entities
+        self._module = module
+
+    def find_organization(self, name):
+        org = self._entities.Organization(self._server, name=name)
+        response = org.search(set(), {'search': 'name={}'.format(name)})
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No organization found for %s" % name)
+
+    def find_lifecycle_environment(self, name, organization):
+        env = self._entities.LifecycleEnvironment(self._server, name=name,
+                                                  organization=organization)
+        response = env.search()
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No lifecycle environment found for %s" % name)
+
+    def find_content_view(self, name, organization):
+        cv = self._entities.ContentView(self._server, name=name,
+                                        organization=organization)
+        response = cv.search()
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No content view found for %s" % name)
+
+    def activation_key(self, name, organization, lifecycle_environment=None, content_view=None):
+        updated = False
+        org = self.find_organization(organization)
+
+        kwargs = {'name': name, 'organization': org}
+
+        if lifecycle_environment:
+            kwargs['environment'] = self.find_lifecycle_environment(lifecycle_environment, org)
+
+        if content_view:
+            kwargs['content_view'] = self.find_content_view(content_view, org)
+
+        activation_key = self._entities.ActivationKey(self._server, **kwargs)
+        response = activation_key.search()
+
+        if len(response) == 1:
+            activation_key = response[0]
+        elif len(response) == 0:
+            activation_key = activation_key.create()
```

Comment:
This only does a create and not an update. If the name of the activation key does not change, but I change the content view or lifecycle environment, I want that change to result in an 'update' to the activation key.